nselib/url.lua

```diff
@@ -173,12 +177,31 @@ function parse(url, default)
     return ""
   end)
   -- get params
-  url = string.gsub(url, "%;(.*)", function(p)
```

Comment:
You're not handling URI params properly here, but the original didn't either. It's a tricky subject: URI params can happen within a path or at the end, and are considered part of the path. They are distinct from the query string. This is a valid URI: `http://example.com/path;param1=foo/path2;param2?query1=bar`. Furthermore, the [W3C recommends](https://www.w3.org/TR/html4/appendix/notes.html#h-B.2.2) that semicolons be allowed instead of ampersands to separate query parameters.

Since no code currently uses the params key (except to rebuild a parsed URL), it's probably best to leave it out and document that URI params are not currently parsed. We can add it later if we choose.